datafusion/core/tests/physical_optimizer/enforce_distribution.rs

```diff
@@ -665,8 +714,12 @@ fn multi_hash_joins() -> Result<()> {
                             "DataSourceExec: file_groups={1 group: [[x]]}, projection=[a, b, c, d, e], file_type=parquet",
                         ],
                 };
-                assert_optimized!(expected, top_join.clone(), true);
-                assert_optimized!(expected, top_join, false);
+                assert_optimized!(
+                    expected,
+                    top_join.clone(),
+                    &TestConfig::new(DoFirst::Distribution)
+                );
```

Comment:
Taking this a step more, what do you think of using a function on `TestConfig` so this looks like:



```suggestion
                TestConfig::new(DoFirst::Distribution)
                  .assert_optimized)
                    &expected,
                    &top_join)
                );
```

Reply:
I like this idea. I was already planning for the next followup PR to change the macro to a functional call.

Reply:
Here is the draft of the followup PR: https://github.com/influxdata/arrow-datafusion/pull/61

Just to get an idea of where this is heading.